static/js/emoji_picker.js

```diff
@@ -232,15 +232,28 @@ function filter_emojis() {
                 continue;
             }
             const emojis = category.emojis;
-
             for (const emoji_dict of emojis) {
+                let match = false;
                 for (const alias of emoji_dict.aliases) {
-                    const match = search_terms.every((search_term) => alias.includes(search_term));
+                    match = search_terms.every((search_term) => alias.includes(search_term));
                     if (match) {
                         search_results.push({...emoji_dict, emoji_name: alias});
                         break; // We only need the first matching alias per emoji.
                     }
                 }
+
+                if (!match && emoji_dict.emoji_code) {
+                    // convert emoji_dict to an actual emoji character
+                    const parsed_emoji_code = typeahead.parse_unicode_emoji_code(
+                        emoji_dict.emoji_code,
+                    );
+                    // using query instead of search_terms because it's possible multiple emojis were input
+                    // without being separated by spaces
+                    match = query.includes(parsed_emoji_code);
```

Comment:
It might be easier to use a different variable name, like `unicode_emoji_match`, here, rather than changing `const` above.

Reply:
Yes, you are right, I initially used the same variable because I only wanted to match the literal emoji iff the text query didn't already match anything. But on second thought, it makes more sense that we still go ahead to match literal emojis regardless of a previous match.

So searching for something like `"sun 🌚"` should still work to match the following result `☀️ 🌚`